integration_tests/ctesque/src/sharedTest/java/android/net/http/HttpEngineTest.java

```diff
@@ -0,0 +1,49 @@
+package android.net.http;
+
+import android.content.Context;
+import android.os.Build;
+import androidx.test.core.app.ApplicationProvider;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.filters.SdkSuppress;
+import java.net.URL;
+import java.net.URLConnection;
+import org.chromium.net.impl.JavaCronetProvider;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.annotation.Config;
+import org.robolectric.annotation.internal.DoNotInstrument;
+import org.robolectric.shadows.ShadowHttpEngine;
+
+@DoNotInstrument
+@RunWith(AndroidJUnit4.class)
+@Config(minSdk = Build.VERSION_CODES.UPSIDE_DOWN_CAKE)
+@SdkSuppress(minSdkVersion = Build.VERSION_CODES.UPSIDE_DOWN_CAKE)
+public class HttpEngineTest {
+  @Test
+  public void supportHttpClientOpenConnection() throws Exception {
+    Context context = ApplicationProvider.getApplicationContext();
+
+    HttpEngine engine = new HttpEngine.Builder(context).build();
+
+    URLConnection con = engine.openConnection(new URL("https://google.com/robots.txt"));
+
+    // Check how to test network in robolectric builds
+    //    String response = new String(con.getInputStream().readAllBytes());
+    //    assertThat(response).contains("Disallow");
+  }
+
+  @Test
+  public void supportsCustomProvider() throws Exception {
+    Context context = ApplicationProvider.getApplicationContext();
+
+    ShadowHttpEngine.setCronetEngineBuilder(new JavaCronetProvider(context).createBuilder());
```

Comment:
@utzcoz does this look right for a shadow.  I hope this lets us avoid dealing with the platform issues, apps can BYO their own embedded provider.

Reply:
Should I leave the existing test here and move the new shadow test?

Or make it private for now and follow up with a separate shadow PR?

Reply:
> If we use explicit `Shadow*` APIs, related tests should be moved to shadow module's test directory. The `ctesque` only places tests that can run on both Robolectric environment and Emulator environment.

Done. Removed for now.